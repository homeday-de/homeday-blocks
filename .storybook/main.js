const path = require('path')
const generateWebpackAlias = require('../webpack-alias');

module.exports = {
  "stories": [
    "../src/**/*.stories.@(js|jsx|ts|tsx|mdx)",
  ],
  "addons": [
    "@storybook/addon-links",
    "@storybook/addon-essentials",
    "@storybook/addon-knobs",
  ],
  webpackFinal: async (config) => {
    config.resolve.alias = generateWebpackAlias({
      custom: {
        '@': './src',
        vue$: './node_modules/vue/dist/vue.esm.js',
      },
    });

    config.module.rules.push({
      test: /\.scss$/,
      use: ['style-loader', 'css-loader', 'sass-loader'],
      include: path.resolve(__dirname, '../'),
    });

    return config;
  },
}
