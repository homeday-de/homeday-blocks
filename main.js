export { default as HdAlert, TYPES as HdAlertTypes } from './src/components/HdAlert.vue';
export { default as HdBadge, TYPES as HdBadgeTypes } from './src/components/HdBadge.vue';
export { default as HdCalendar } from './src/components/HdCalendar.vue';
export { default as HdEditSwitch } from './src/components/HdEditSwitch.vue';
export { default as HdExpandText } from './src/components/HdExpandText.vue';
export { default as HdLazyImage } from './src/components/HdLazyImage.vue';
export { default as HdLink, TYPES as HdLinkTypes } from './src/components/HdLink.vue';
export { default as HdPager } from './src/components/HdPager.vue';
export { default as HdResponsive } from './src/components/HdResponsive.vue';
export { default as HdTable } from './src/components/HdTable.vue';
export { default as HdTabsMenu } from './src/components/HdTabsMenu.vue';
export { default as HdTagsList } from './src/components/HdTagsList.vue';
export { default as HdTimeslots } from './src/components/HdTimeslots.vue';
export { default as HdToast } from './src/components/HdToast.vue';
export { default as HdIcon } from './src/components/HdIcon.vue';

// Buttons
export { default as HdArrowButton } from './src/components/buttons/HdArrowButton.vue';
export { default as HdButton, TYPES as HdButtonTypes } from './src/components/buttons/HdButton.vue';
export { default as HdLoaderButton } from './src/components/buttons/HdLoaderButton.vue';
export { default as HdRadioButton } from './src/components/buttons/HdRadioButton.vue';

// Table
export { default as HdDetailsTable } from './src/components/details-table/HdDetailsTable.vue';
export { default as HdDetailsTableRow } from './src/components/details-table/HdDetailsTableRow.vue';

// Form
export { default as HdAutocomplete } from './src/components/form/HdAutocomplete.vue';
export { default as HdCheckbox } from './src/components/form/HdCheckbox.vue';
export { default as HdDynamicForm } from './src/components/form/HdDynamicForm.vue';
export { default as HdGoogleAutocomplete } from './src/components/form/HdGoogleAutocomplete.vue';
export { default as HdInput } from './src/components/form/HdInput.vue';
export { default as HdInputPassword } from './src/components/form/HdInputPassword.vue';
export { default as HdPasswordConfirm } from './src/components/form/HdPasswordConfirm.vue';
export { default as HdRadio } from './src/components/form/HdRadio.vue';
export { default as HdRange } from './src/components/form/HdRange.vue';
export { default as HdSelect } from './src/components/form/HdSelect.vue';
export { default as HdSplitInput } from './src/components/form/HdSplitInput.vue';
export { default as HdTagsSelector } from './src/components/form/HdTagsSelector.vue';
export { default as HdTextarea } from './src/components/form/HdTextarea.vue';

// Tooltip
export { default as HdTooltip } from './src/components/tooltip/HdTooltip.vue';
export { default as HdTooltipped } from './src/components/tooltip/HdTooltipped.vue';
export { default as HdTooltipInstaller } from './src/components/tooltip/installer';

// Gallery
export { default as HdGallery } from './src/components/gallery/HdGallery.vue';
export { default as HdGalleryCarousel } from './src/components/gallery/HdGalleryCarousel.vue';
export { default as HdGalleryMedia } from './src/components/gallery/HdGalleryMedia.vue';
export { default as HdGalleryOverlay } from './src/components/gallery/HdGalleryOverlay.vue';
export { default as HdGalleryPlaceholder } from './src/components/gallery/HdGalleryPlaceholder.vue';
export { default as HdGalleryTiles } from './src/components/gallery/HdGalleryTiles.vue';
export { default as HdZoomerGallery } from './src/components/gallery/HdZoomerGallery.vue';

// Notifications
export { default as HdNotifications } from './src/components/notifications/HdNotifications.vue';
export { default as HdNotificationsBar } from './src/components/notifications/HdNotificationsBar.vue';

// Services
export { default as BreakpointsService } from './src/services/breakpoints';
export { default as DateService } from './src/services/date';
export { default as EventEmitterService } from './src/services/event-emitter';
export { default as FlickityService } from './src/services/flickity';
export { default as FormValidationService } from './src/services/formValidation';
export { default as GoogleApiService } from './src/services/googleAPI';
export { default as OnResizeService } from './src/services/on-resize';
export { default as ScrollingService } from './src/services/scrolling';
export { default as UtilsService } from './src/services/utils';
