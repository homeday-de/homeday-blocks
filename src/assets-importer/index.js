/* eslint-disable */
require('dotenv').config();
const fs = require('fs');
const fetch = require('node-fetch');
const _camelCase = require('lodash/camelCase');
const CONFIG = require('./CONFIG');

if (process.env.FIGMA_ACCESS_TOKEN === undefined) {
  console.error('FIGMA_ACCESS_TOKEN is undefined. Did you add it to your environment variables?');
  return;
}

function fetchFromFigma(endpoint) {
  return fetch(`https://api.figma.com/v1${endpoint}`, {
    headers: {
      'x-FIGMA-token': process.env.FIGMA_ACCESS_TOKEN,
    },
  }).then(res => res.json());
}

function normalizeFilename({ name, regex }) {
  const matches = name.match(regex);

  return matches ? matches.pop() : null;
}

async function download({
  url,
  filename,
  dist = '',
}) {
  const distPath = `${process.cwd()}${dist}`;
  await fs.mkdir(distPath, { recursive: true }, (err) => {
    if (err) throw err;
  });
  const writer = fs.createWriteStream(`${distPath}/${filename}`);

  const response = await fetch(url);

  response.body.pipe(writer);

  return new Promise((resolve, reject) => {
    writer.on('finish', resolve(filename));
    writer.on('error', reject);
  });
}

async function importAssetsFromFigma({
  fileKey,
  format,
  dist,
  filenameRegex,
  shouldMatchRegex,
}) {
  const file = await fetchFromFigma(`/files/${fileKey}/components`);

  if (!file.meta.components || file.meta.components.length === 0) {
    console.warn('No components found for file', fileKey);
    return;
  }

  const filenameMap = file.meta.components
    .reduce((map, component) => {
      const isMatch = Array.isArray(component.name.match(shouldMatchRegex));
      if (isMatch) {
        const filename = normalizeFilename({ name: component.name, regex: filenameRegex });
        map[component.node_id] = filename;
      }

      return map;
    }, {});

  console.log('filenameMap', filenameMap);

  const ids = Object.keys(filenameMap).join(',');
  const assets = await fetchFromFigma(`/images/${fileKey}?ids=${ids}&format=${format}`);

  if (assets.images === undefined) {
    console.warn('No images found for regexp', shouldMatchRegex);
    return;
  }

  const downloadedFiles = await Promise.all(
    Object.entries(assets.images)
      .map(([id, url]) => url && download({
        url,
        filename: `${filenameMap[id]}.${format}`,
        dist,
      })),
  );

  createIndex({
    files: downloadedFiles,
    dist,
    exportNameRegex: new RegExp(`ic_(.*).${format}`),
    suffix: 'Icon',
  });
}

function createIndex({ files, dist, exportNameRegex = '*', suffix = '' }) {
  const fileContent = [...new Set(files)]
    .reduce ((content, file) => {
      const nameMatches = file.match(exportNameRegex);
      if (!nameMatches) {
        return content;
      }

      const exportName = _camelCase(nameMatches.pop()) + suffix;
      return `${content}\nexport { default as ${exportName} } from './${file}';\n`;
    }, '');

  console.log('fileContent', fileContent);

  fs.writeFile(`${process.cwd()}${dist}/index.js`, fileContent, (err) => {
    if (err) console.error('Failed to create index', err);
  });
}

CONFIG.FIGMA_FILES.forEach(FIGMA_FILE => importAssetsFromFigma({
  fileKey: FIGMA_FILE.KEY,
  format: FIGMA_FILE.FORMAT,
  filenameRegex: FIGMA_FILE.FILENAME_REGEX,
  shouldMatchRegex: FIGMA_FILE.SHOULD_MATCH_REGEX,
  dist: FIGMA_FILE.DIST,
}));
