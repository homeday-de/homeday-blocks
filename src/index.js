// This barrel module has been used to export all the non-compiled/raw components and services to be used in Nuxt SSR/SSG projects.

// Namespaces
export { default as HdLinkTypes } from './components/HdLinkTypes';
export { default as HdAlertTypes } from './components/HdAlertTypes';
export { default as HdBadgeTypes } from './components/HdBadgeTypes';
export { default as HdButtonTypes } from './components/buttons/HdButtonTypes';
export { default as HdTagTypes } from './components/HdTagTypes';
export { default as HdNotificationsTypes } from './components/notifications/HdNotificationsTypes';

// Components
export { default as HdAlert } from './components/HdAlert.vue';
export { default as HdBadge } from './components/HdBadge.vue';
export { default as HdCalendar } from './components/HdCalendar.vue';
export { default as HdEditSwitch } from './components/HdEditSwitch.vue';
export { default as HdExpandText } from './components/HdExpandText.vue';
export { default as HdLazyImage } from './components/HdLazyImage.vue';
export { default as HdLink } from './components/HdLink.vue';
export { default as HdPager } from './components/HdPager.vue';
export { default as HdResponsive } from './components/HdResponsive.vue';
export { default as HdTable } from './components/HdTable.vue';
export { default as HdTabsMenu } from './components/HdTabsMenu.vue';
export { default as HdTagsList } from './components/HdTagsList.vue';
export { default as HdTimeslots } from './components/HdTimeslots.vue';
export { default as HdToast } from './components/HdToast.vue';
export { default as HdToggle } from './components/HdToggle.vue';
export { default as HdIcon } from './components/HdIcon.vue';
export { default as HdModal } from './components/HdModal.vue';
export { default as HdDashedList } from './components/HdDashedList.vue';
export { default as HdLoadingSpinner } from './components/HdLoadingSpinner.vue';

// Buttons
export { default as HdArrowButton } from './components/buttons/HdArrowButton.vue';
export { default as HdButton } from './components/buttons/HdButton.vue';
export { default as HdLoaderButton } from './components/buttons/HdLoaderButton.vue';
export { default as HdRadioButton } from './components/buttons/HdRadioButton.vue';

// Table
export { default as HdDetailsTable } from './components/details-table/HdDetailsTable.vue';
export { default as HdDetailsTableRow } from './components/details-table/HdDetailsTableRow.vue';

// Form
export { default as HdCheckbox } from './components/form/HdCheckbox.vue';
export { default as HdCheckboxCard } from './components/form/HdCheckboxCard.vue';
export { default as HdCheckboxCardGroup } from './components/form/HdCheckboxCardGroup.vue';
export { default as HdDynamicForm } from './components/form/HdDynamicForm.vue';
export { default as HdForm } from './components/form/HdForm.vue';
export { default as HdGoogleAutocomplete } from './components/form/HdGoogleAutocomplete.vue';
export { default as HdInput } from './components/form/HdInput.vue';
export { default as HdInputFormatter } from './components/form/HdInputFormatter.vue';
export { default as HdInputPassword } from './components/form/HdInputPassword.vue';
export { default as HdPasswordConfirm } from './components/form/HdPasswordConfirm.vue';
export { default as HdRadio } from './components/form/HdRadio.vue';
export { default as HdRadioCard } from './components/form/HdRadioCard.vue';
export { default as HdRadioCardGroup } from './components/form/HdRadioCardGroup.vue';
export { default as HdRange } from './components/form/HdRange.vue';
export { default as HdSelect } from './components/form/HdSelect.vue';
export { default as HdSplitInput } from './components/form/HdSplitInput.vue';
export { default as HdTagsSelector } from './components/form/HdTagsSelector.vue';
export { default as HdTextarea } from './components/form/HdTextarea.vue';
export { default as HdTileSelect } from './components/form/HdTileSelect.vue';

// Tooltip
export { default as HdTooltip } from './components/tooltip/HdTooltip.vue';
export { default as HdTooltipped } from './components/tooltip/HdTooltipped.vue';
export { default as HdTooltipInstaller } from './components/tooltip/HdTooltipInstaller';

// Gallery
export { default as HdGallery } from './components/gallery/HdGallery.vue';
export { default as HdGalleryCarousel } from './components/gallery/HdGalleryCarousel.vue';
export { default as HdGalleryMedia } from './components/gallery/HdGalleryMedia.vue';
export { default as HdGalleryOverlay } from './components/gallery/HdGalleryOverlay.vue';
export { default as HdGalleryPlaceholder } from './components/gallery/HdGalleryPlaceholder.vue';
export { default as HdGalleryTiles } from './components/gallery/HdGalleryTiles.vue';
export { default as HdZoomerGallery } from './components/gallery/HdZoomerGallery.vue';

// Notifications
export { default as HdNotifications } from './components/notifications/HdNotifications.vue';
export { default as HdNotification } from './components/notifications/HdNotification.vue';
export { default as HdInputPhone } from './components/form/HdInputPhone.vue';

// Services
export { default as onResizeService } from './services/on-resize';
export { default as EventEmitter } from './services/event-emitter';
export * from './services/breakpoints';
export * from './services/date';
export * from './services/flickity';
export * from './services/formValidation';
export * from './services/googleAPI';
export * from './services/scrolling';
export * from './services/utils';
