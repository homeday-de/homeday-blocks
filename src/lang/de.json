{
  "GENERAL": {
    "EDIT": "Bearbeiten",
    "SAVE": "Speichern",
    "CANCEL": "Abbrechen",
    "SAVING": "Speichern"
  },
  "FORM": {
    "EMAIL": {
      "PLACEHOLDER": "z.b. michael.schmidt@gmail.com",
      "LABEL": "Ihre E-Mail-Adresse"
    },
    "PASSWORD": {
      "LABEL": "Ihr Passwort",
      "LABEL_CONFIRM": "Bestätigen Sie Ihr Passwort"
    },
    "VALIDATION": {
      "REQUIRED": "Dies ist ein Pflichtfeld.",
      "INVALID_EMAIL": "Bitte geben Sie eine korrekte E-Mail-Adresse ein.",
      "INVALID_DATE": "Bitte geben Sie ein Datum in diesem Format ein: JJJJ-MM-TT",
      "PASSWORD_MISMATCH": "Dies stimmt nicht mit dem obigen Passwort überein",
      "PASSWORD_SHORT": "Dieses Passwort ist zu kurz",
      "PASSWORD_STRENGTH": {
        "TITLE": "Stärke",
        "LEVELS": [
          "Schwach",
          "Mittlere",
          "Stark",
          "Sehr stark"
        ]
      }
    }
  },
  "EXPAND_TEXT": {
    "TOGGLE": {
      "SHOW_MORE": "weiterlesen",
      "SHOW_LESS": "zuklappen"
    }
  },
  "GALLERY_TILES": {
    "MORE_PHOTOS": "Weitere Fotos"
  }
}
