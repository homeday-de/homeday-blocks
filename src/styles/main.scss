@import './mixins';
@import './buttons';
@import './inputs';

* {
  box-sizing: border-box;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

body {
  @include font('body');
  color: $primary-color;
}


/** SPACING **/
/* CLASSES UTILITIES GENERATION */
$spacing-properties : 'margin', 'padding';

@each $property in $spacing-properties {
  @each $name, $size in $standards {
    .#{$property}-top-#{$name} {
      #{$property}-top: $size;
    }
    .#{$property}-right-#{$name} {
      #{$property}-right: $size;
    }
    .#{$property}-bottom-#{$name} {
      #{$property}-bottom: $size;
    }
    .#{$property}-left-#{$name} {
      #{$property}-left: $size;
    }
    .#{$property}-#{$name} {
      #{$property}: $size;
    }
  }
}

/* MIX-IN WITH SAME PURPOSE THAN CLASSES UTILITIES */
$list-position: 'top', 'bottom', 'right', 'left';
@mixin margin($postion, $value) {
  //Check if position exist
  @if $postion != '' {
    //First level position compare
    @each $pos in $list-position {
      @if $pos == $postion {
        //Second level standards check
        @each $name, $size in $standards {
          @if $value == $name {
            margin-#{$pos}: $size;
          }
        }
      }
    }
  } @else {
    //Second level standards check for all side
    @each $name, $size in $standards {
      @if $value == $name {
        margin: $size;
      }
    }
  }
}
@mixin padding($postion, $value) {
  //Check if position exist
  @if $postion != '' {
    //First level position compare
    @each $pos in $list-position {
      @if $pos == $postion {
        //Second level standards check
        @each $name, $size in $standards {
          @if $value == $name {
            padding-#{$pos}: $size;
          }
        }
      }
    }
  } @else {
    //Second level standards check for all side
    @each $name, $size in $standards {
      @if $value == $name {
        padding: $size;
      }
    }
  }
}


/** FONTS **/

/* GENERATION OF UTILITIES CLASSES */

$font-list: ('headline', 'headline'),
('subhead', 'subhead'),
('title', 'title'),
('subtitle', 'subtitle'),
('xsheadline', 'xsheadline'),
('body', 'body'),
('body--primary', 'body--primary'),
('body--secondary', 'body--secondary'),
('text-small', 'text-small'),
('text-xsmall', 'text-xsmall'),
('text-xxsmall', 'text-xxsmall'),
('showcase-link', 'showcase-link'),
('link', 'link'),
('link--primary', 'link--primary'),
('link--secondary', 'link--secondary'),
('quote', 'quote'),
('number', 'number'),
('table', 'table'),
('code', 'code');
@each $class-name, $font-name in $font-list {
  .#{$class-name} {
    @include font($font-name);
  }
}
