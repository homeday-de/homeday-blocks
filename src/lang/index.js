// @ts-check
import de from './de.json';
import en from './en.json';

/**
 * @typedef {{ [key in keyof de]: de[key] }} Messages
 */

const langs = {
  de,
  en,
};

/**
 * Get static Homeday Blocks translation files
 *
 * @param {string} lang
 * @returns {Messages}
 */
export function getMessages(lang) {
  return langs[lang] || langs.de;
}

export default {
  getMessages,
};
