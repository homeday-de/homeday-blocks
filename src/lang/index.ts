import de from './de.json';
import en from './en.json';

export type Messages = { [key in keyof typeof de]: typeof de[key] };

const langs = {
  de,
  en,
};

/**
 * Get static Homeday Blocks translation files
 */
export function getMessages(lang: string): Messages {
  return langs[lang] || langs.de;
}

export default {
  getMessages,
};
