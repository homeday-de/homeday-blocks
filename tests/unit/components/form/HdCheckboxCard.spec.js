// @ts-check

import { mount } from '@vue/test-utils';
import deepmerge from 'deepmerge';
import { HdCheckboxCard } from 'homeday-blocks/main';
import { getMessages } from 'homeday-blocks/src/lang';

describe('HdCheckboxCard', () => {
  const build = (overrideProps = {}, overrideSlots = {}) => {
    const propsData = deepmerge({ name: 'property', nativeValue: 'house' }, overrideProps);

    const slots = deepmerge({
      default: ['House'],
      icon: ['<HdIcon src="house-icon" />'],
    }, overrideSlots);

    const view = mount(HdCheckboxCard, {
      propsData,
      slots,
      stubs: {
        HdIcon: true,
      },
    });

    return {
      view,
      input: () => view.find('input[type="checkbox"]'),
      label: () => view.find('label'),
      error: () => view.find('.error'),
      /** @returns {import('homeday-blocks/src/lang').Messages} */
      t: (lang = 'de') => getMessages(lang),
    };
  };

  it('renders correctly', () => {
    const { view } = build();
    expect(view.html()).toMatchSnapshot();
  });

  it('displays a disabled checkbox', async () => {
    const { view, input } = build({ disabled: true });

    await input().setChecked();

    expect(input().element).toBeDisabled();
    expect(view.emitted().input).toBeFalsy();
  });

  it('displays a checkbox with a checked state', () => {
    const { input } = build({ nativeValue: 'house', value: ['house'] });
    expect(input().element).toBeChecked();
  });

  it('displays a checkbox with an unchecked state', () => {
    const { input } = build({ nativeValue: 'house', value: ['apartment'] });
    expect(input().element).not.toBeChecked();
  });

  it('displays validation error when input is required', async () => {
    const { view, error, t } = build({ required: true });

    expect(error().exists()).toBeFalsy();

    await view.setProps({
      value: null,
    });

    expect(error().exists()).toBeTruthy();
    expect(error().text()).toContain(t().FORM.VALIDATION.SELECT_ONE_OPTION);
  });

  it('displays validation error for custom validation rules', async () => {
    const customRules = [
      {
        validate: (value) => value.length > 1,
        errorMessage: 'Value must be greater than 1',
      },
      {
        validate: (value) => value.length < 3,
        errorMessage: 'Value must be less than 3',
      },
    ];
    const { view, error } = build({ customRules });

    expect(error().exists()).toBeFalsy();

    await view.setProps({
      value: 'a',
    });

    expect(error().exists()).toBeTruthy();
    expect(error().text()).toContain(customRules[0].errorMessage);

    await view.setProps({
      value: 'abc',
    });

    expect(error().exists()).toBeTruthy();
    expect(error().text()).toContain(customRules[1].errorMessage);
  });

  describe('emits input event', () => {
    it('when input is checked', async () => {
      const { view, input } = build();

      await input().setChecked();

      expect(view.emitted().input).toBeTruthy();
      expect(view.emitted().input).toHaveLength(1);
    });

    it('when user press space in label', async () => {
      const { view, label } = build();

      await label().trigger('keydown.space');

      expect(view.emitted().input).toBeTruthy();
      expect(view.emitted().input).toHaveLength(1);
    });

    it('for custom true-value', async () => {
      const { view, input } = build({ nativeValue: null, trueValue: 'on' });

      await input().setChecked();

      expect(view.emitted().input).toBeTruthy();
      expect(view.emitted().input).toHaveLength(1);
      expect(view.emitted().input).toEqual([['on']]);
    });

    it('for custom false-value', async () => {
      const { view, input } = build({
        nativeValue: null, value: 'on', trueValue: 'on', falseValue: 'off',
      });

      await input().setChecked(false);

      expect(view.emitted().input).toBeTruthy();
      expect(view.emitted().input).toHaveLength(1);
      expect(view.emitted().input).toEqual([['off']]);
    });
  });
});
