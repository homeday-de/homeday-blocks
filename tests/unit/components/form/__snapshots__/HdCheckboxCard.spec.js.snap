// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HdCheckboxCard renders correctly 1`] = `
<label for="property-house" tabindex="0"><input type="checkbox" id="property-house" name="property" true-value="true" value="house">
  <div class="card">
    <div class="card__border"></div>
    <div class="checkbox-indicator card__indicator">
      <div class="checkbox-indicator__box">
        <!---->
      </div>
    </div>
    <div class="card__icon">
      <hdicon-stub src="house-icon"></hdicon-stub>
    </div> <span class="card__control-label">House</span>
  </div>
  <!---->
</label>
`;
